pkg/ca/kmsca/kmsca.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package kmsca
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"os"
+	"path/filepath"
+
+	"github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/ca/intermediateca"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/signature/kms"
+
+	// Register the provider-specific plugins
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/aws"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/azure"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/gcp"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/hashivault"
```

Comment:
Nit: I prefer having imports with side effects (e.g loading database drivers and plugins etc) in `package main`. You can see an example of this behaviour [here in `distribution/distribution`](https://github.com/distribution/distribution/blob/main/cmd/registry/main.go). This makes it really easy for folks to build a slimmed down version of the registry (or fulcio in our case) by just customizing that `package main` themselves. 

Example

```golang
package main

import (
	"github.com/distribution/distribution/v3/registry"

	_ "github.com/distribution/distribution/v3/registry/auth/token"          # Only token auth
	_ "github.com/distribution/distribution/v3/registry/storage/driver/gcs"  # Only GCS storage
)

func main() {
	registry.RootCmd.Execute()
}
```

Reply:
Interesting point! I feel like there's a larger thing here - If you want to create a slimmed down Fulcio, you should be able to remove all signer types that aren't in use. Given this is the KMS type, I'd prefer that we load in all supported KMS backends in the same file.

Reply:
Open to making that file be in `package main`? I think I see what you're saying in that we should be able to turn on and off all of "kmsca" with an import drop or configuration but I think its nice to also be able to say e.g "I was _kmsca_ and specifically only support gcpkms:// keys" and that still isn't possible if these driver imports happen under /pkg

Reply:
The benefit of moving these out to `main` would be if someone wanted to use `kmsca` as a library, correct? I don't expect that to be very common, so I'd rather simply install all of the supported KMS backends so if someone does use `kmsca` without `main`, they won't run into issues with missing KMS backends.

Reply:
The benefit is in being able to have fine grain control of the `kmsca` backends supported by dropping them in main instead of using `kmsca` as a library, but I see your point about using `kmsca` as a lib. I'm alright to have coarse control of the entire kmsca backend in `package main`